spring-cloud-data-core/src/main/java/org/springframework/cloud/data/core/parser/StreamDefinitionParser.java

```diff
@@ -0,0 +1,124 @@
+/*
+ * Copyright 2015 the original author or authors.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ * http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package org.springframework.cloud.data.core.parser;
+
+
+import java.util.ArrayList;
+import java.util.Deque;
+import java.util.LinkedList;
+import java.util.List;
+
+import org.springframework.cloud.data.core.ModuleDefinition;
+import org.springframework.cloud.data.core.dsl.ArgumentNode;
+import org.springframework.cloud.data.core.dsl.ModuleNode;
+import org.springframework.cloud.data.core.dsl.SinkChannelNode;
+import org.springframework.cloud.data.core.dsl.SourceChannelNode;
+import org.springframework.cloud.data.core.dsl.StreamDslParser;
+import org.springframework.cloud.data.core.dsl.StreamNode;
+
+/**
+ * Parser to convert a DSL string for a stream into a list of
+ * {@link ModuleDefinition} objects that comprise the given stream.
+ *
+ * @author Andy Clement
+ * @author Gunnar Hillert
+ * @author Glenn Renfro
+ * @author Mark Fisher
+ * @author Patrick Peralta
+ * @author Eric Bottard
+ * @since 1.0
+ */
+public class StreamDefinitionParser {
```

Comment:
I wonder if we should call this `ModuleDefinitionParser`, and likewise the `StreamDslParser` could be renamed `StreamNodeParser`. That way both of them are named for what they return as a result of parsing.